PollBuddy-Server/backend/routes/users.test.js

```diff
@@ -0,0 +1,545 @@
+      userData[key] = value;
+    }
+  }
+  
+  let user = createModel(userData, userSchema);
+  let res = await mongoConnection.getDB().collection("users").insertOne(user);
+  return res;
+};
+
+describe("/api/users/login", () => {
+
+  it("GET: fail", async () => {
+    await app.get("/api/users/login")
+      .expect(405)
+      .then((response) => {
+        expect(response.body.result).toBe("failure");
+      });
+  });
+
+  it("POST: login with username success", async () => {
+    let res = await createUser();
+    let userLogin = {
+      userNameEmail: testUser.UserName,
+      password: testUser.Password,
+    };
+
+    await app.post("/api/users/login")
+      .send(userLogin)
+      .expect(200)
+      .then((response) => {
+        expect(response.body.result).toBe("success");
+        expect(response.body.data.firstName).toBe(testUser.FirstName);
+        expect(response.body.data.lastName).toBe(testUser.LastName);
+        expect(response.body.data.userName).toBe(testUser.UserName);
+        expect(session).toHaveProperty("userData.userID", res.insertedId);  
+      });
+  });
+
+  it("POST: login with email success", async () => {
+    let res = await createUser();
+    let userLogin = {
+      userNameEmail: testUser.UserName,
+      password: testUser.Password,
+    };
+
+    await app.post("/api/users/login")
+      .send(userLogin)
+      .expect(200)
+      .then((response) => {
+        expect(response.body.result).toBe("success");
+        expect(response.body.data.firstName).toBe(testUser.FirstName);
+        expect(response.body.data.lastName).toBe(testUser.LastName);
+        expect(response.body.data.userName).toBe(testUser.UserName);
+        expect(session).toHaveProperty("userData.userID", res.insertedId);  
+      });
+  });
+
+});
+
+describe("/api/users/register", () => {
+  
+  it("GET: fail", async () => {
+    await app.get("/api/users/register")
+      .expect(405)
+      .then((response) => {
+        expect(response.body.result).toBe("failure");
+      });
+  });
+
+  it("POST: register user success", async () => {
+    await app.post("/api/users/register")
+      .send({
+        userName: testUser.UserName,
+        email: testUser.Email,
+        password: testUser.Password,
+        firstName: testUser.FirstName,
+        lastName: testUser.LastName
+      })
+      .expect(200)
+      .then(async (response) => {
+        expect(response.body.result).toBe("success");
+        expect(response.body.data.firstName).toBe(testUser.FirstName);
+        expect(response.body.data.lastName).toBe(testUser.LastName);
+        expect(response.body.data.userName).toBe(testUser.UserName);
+
+        let res = await mongoConnection.getDB().collection("users").findOne({ 
+          UserName: testUser.UserName,
+          Email: testUser.Email,
+          FirstName: testUser.FirstName,
+          LastName: testUser.LastName
+        });
+
+        expect(res).toBeTruthy();
```

Comment:
Double check that res is false if findOne doesn't actually find anything

Reply:
yea test fails if it doesn't find anything.